koku/reporting_common/models.py

```diff
@@ -4,8 +4,42 @@
 #
 """Models for shared reporting tables."""
 from django.db import models
+from django.db.models import IntegerChoices
 from django.utils import timezone
 
+from common.enum import StrEnum
+
+
+class ReportStep(IntegerChoices):
+    DOWNLOADING = 3
+    PROCESSING = 4
+    OCP_CLOUD_PROCESSING = (5, "OCP-Cloud-Processing")
+
+
+class Status(IntegerChoices):
+    DONE = 1
+    FAILED = 2
+
+
+class CombinedChoices(IntegerChoices):
+    DOWNLOADING = ReportStep.DOWNLOADING
+    PROCESSING = ReportStep.PROCESSING
+    OCP_CLOUD_PROCESSING = ReportStep.OCP_CLOUD_PROCESSING
+    DONE = Status.DONE
+    FAILED = Status.FAILED
+
+
+class ManifestStep(StrEnum):
+    DOWNLOAD = "download"
+    PROCESSING = "processing"
+    SUMMARY = "summary"
+
+
+class ManifestState(StrEnum):
+    START = "start"
+    END = "end"
+    FAILED = "failed"
```

Comment:
Code organization nit: It kind seems like these should be in the `common` module. Maybe `common.status` or `common.state`? It seems odd having supporting classes that are not models in `reporting_common.models`.

Side note: `IntegrerChoices` isn't actually a `Model` but an `Enum`.

```
IntegerChoices.__mro__
(<enum 'IntegerChoices'>, int, <enum 'Choices'>, <enum 'Enum'>, object)
```

It's just in `django.db.models` for convenience.
```
IntegerChoices.__class__
django.db.models.enums.ChoicesMeta
```